lib/trino-hive-formats/src/main/java/io/trino/hive/formats/avro/AvroFileReader.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.hive.formats.avro;
+
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.hive.formats.TrinoDataInputStream;
+import org.apache.avro.Schema;
+import org.apache.avro.file.SeekableInput;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+import static com.google.common.base.Verify.verify;
+import static java.util.Objects.requireNonNull;
+
+public class AvroFileReader
+        implements Closeable
+{
+    private final String filePath;
+    private final TrinoDataInputStream input;
+    private final AvroFilePageIterator pageIterator;
+
+    public AvroFileReader(
```

Comment:
I'm not sure this abstraction is necessary.  I'm thinking merge `AvroFilePageIterator` into this class.  Then we are hiding avro details behind this interface

Reply:
done